src/main/java/com/hubspot/jinjava/lib/filter/SliceFilter.java

```diff
@@ -80,20 +80,26 @@ public Object filter(Object var, JinjavaInterpreter interpreter, String... args)
         args[0]
       );
     }
+
     List<List<Object>> result = new ArrayList<>();
+    List<Object> currentList = null; // lazy evaluation
 
-    List<Object> currentList = null;
     int i = 0;
     while (loop.hasNext()) {
-      Object next = loop.next();
       if (i % slices == 0) {
-        currentList = new ArrayList<>(slices);
-        result.add(currentList);
+        if (currentList != null) {
+          result.add(currentList);
+        }
+        currentList = new ArrayList<>();
```

Comment:
```suggestion
        currentList = new ArrayList<>(slices);
```
so we never have to resize the array.

Reply:
as we discussed we keep it due to better heap results